src/OpenTelemetry.Api/Baggage.cs

```diff
@@ -46,21 +46,34 @@ internal Baggage(Dictionary<string, string> baggage)
         /// <summary>
         /// Gets or sets the current <see cref="Baggage"/>.
         /// </summary>
+        /// <remarks>
+        /// Note: <see cref="Current"/> returns a copy of the current Baggage.
+        /// Changes to the returned instance will not automatically be reflected
+        /// back on the <see cref="Current"/> version. To update the <see
+        /// cref="Current"/> version either set <see cref="Current"/> to a new
+        /// instance or use one of the static methods that target <see
+        /// cref="Current"/> as the default source. Examples:
+        /// <code>
+        /// Baggage.SetBaggage("newKey1", "newValue1"); // Updates Baggage.Current with 'newkey1'
+        /// Baggage.SetBaggage("newKey2", "newValue2"); // Updates Baggage.Current with 'newkey2'
+        /// </code>
+        /// Or:
+        /// <code>
+        /// var baggageCopy = Baggage.Current;
+        /// Baggage.SetBaggage("newKey1", "newValue1"); // Updates Baggage.Current with 'newkey1'
+        /// var newBaggage = baggageCopy
+        ///     .SetBaggage("newKey2", "newValue2");
+        ///     .SetBaggage("newKey3", "newValue3");
+        /// /* Sets Baggage.Current to 'newBaggage' which will override any
```

Comment:
nit: maybe change this to line comments (mixing the two types of comments is a bit weird)

```suggestion
        /// // Sets Baggage.Current to 'newBaggage' which will override any
```